3.resistance-signature/scripts/nbconverted/8.1. last-batch-compile-otherclone-datase.py

```diff
@@ -0,0 +1,163 @@
+
+
+import sys
+import pathlib
+import numpy as np
+import pandas as pd
+
+from pycytominer.cyto_utils import infer_cp_features
+
+sys.path.insert(0, "../2.describe-data/scripts")
+from processing_utils import load_data
+
+
+# In[2]:
+
+
+np.random.seed(1234)
+
+
+# In[3]:
+
+
+data_dir = pathlib.Path("../0.generate-profiles/profiles")
+cell_count_dir = pathlib.Path("../0.generate-profiles/cell_counts/")
+
+output_dir = pathlib.Path("data")
+
+profile_suffix = "normalized.csv.gz"
+
+
+# In[4]:
+
+
+datasets_val = {
+        "2021_08_02_Batch24": ["221057"],
+        "2021_08_02_Batch25": ["221058"],
+        "2021_08_03_Batch26": ["221093"],
+        "2021_08_03_Batch27": ["221094"],
+}
+
+
+# In[5]:
+
+
+#added 'val' to each original variable names from 8.0
+
+full_df_val = []
+for dataset_val in datasets_val:
+    dataset_df_val = []
+    for batch in datasets_val:
+        plates = datasets_val[batch]
+        
+        df_val = load_data(
+            batch=batch,
+            plates=plates,
+            profile_dir=data_dir,
+            suffix=profile_suffix,
+            combine_dfs=True,
+            harmonize_cols=True,
+            add_cell_count=True,
+            cell_count_dir=cell_count_dir
+        )
+        
+        # Add important metadata features
+        df_val = df_val.assign(
+            Metadata_dataset=dataset_val,
+            Metadata_batch=batch,
+            Metadata_clone_type="resistant",
+            Metadata_clone_type_indicator=1,
+            Metadata_model_split="otherclone"
+        )
+
+        df_val.loc[df_val.Metadata_clone_number.str.contains("WT"), "Metadata_clone_type"] = "sensitive"
+        df_val.loc[df_val.Metadata_clone_number.str.contains("WT"), "Metadata_clone_type_indicator"] = 0
+        dataset_df_val.append(df_val)
+
+    # Merge plates of the same dataset together
+    dataset_df_val = pd.concat(dataset_df_val, axis="rows", sort=False).reset_index(drop=True)
+    
+    # Generate a unique sample ID
+    # (This will be used in singscore calculation)
+    dataset_df_val = dataset_df_val.assign(
+        Metadata_unique_sample_name=[f"profile_{x}_{dataset_val}" for x in range(0, dataset_df_val.shape[0])]
+    )
+    
+    full_df_val.append(dataset_df_val)
+
+#remove other clone types from the df
+full_df_val = pd.concat(full_df_val, axis="rows", sort=False).reset_index(drop=True)
+full_df_val = full_df_val[full_df_val.Metadata_clone_number != 'WT_parental']
+full_df_val = full_df_val[full_df_val.Metadata_clone_number != 'CloneA']
+full_df_val = full_df_val[full_df_val.Metadata_clone_number != 'CloneE']
+full_df_val = full_df_val[full_df_val.Metadata_clone_number != 'TX clone 2']
+full_df_val = full_df_val[full_df_val.Metadata_clone_number != 'TX clone 3']
+full_df_val = full_df_val[full_df_val.Metadata_clone_number != 'TX clone 4']
+full_df_val = full_df_val[full_df_val.Metadata_clone_number != 'TX clone 5']
+full_df_val = full_df_val[full_df_val.Metadata_clone_number != 'TX clone 6']
```

Comment:
You might need to tinker with this a bit (I didn't test) but it is more [pythonic](https://towardsdatascience.com/how-to-be-pythonic-and-why-you-should-care-188d63a5037e)

remember, the change actually should happen in the `.ipynb` file!

```python
tx_clones = [
    "TX clone 2",
    "TX clone 3",
    "TX clone 4",
    "TX clone 5",
    "TX clone 6",
]
full_df_val = full_df_val.query("Metadata_clone_number not in @tx_clones")
```

Reply:
Thank you Greg! 
I tried `full_df_val[~full_df_val.Metadata_clone_number.str.startswith('TX')]` and few other similar commands but did not work for some reasons.